modules/spotxBidAdapter.js

```diff
@@ -0,0 +1,343 @@
+            cpm: spotxBid.price,
+            creativeId: spotxBid.crid || '',
+            ttl: 360,
+            netRevenue: true,
+            channel_id: serverResponseBody.id,
+            cache_key: spotxBid.ext.cache_key,
+            video_slot: request.video.ext.videoSlot
+          };
+
+          if (request.video) {
+            bid.vastUrl = '//search.spotxchange.com/ad/vast.html?key=' + spotxBid.ext.cache_key;
+            bid.mediaType = VIDEO;
+            bid.width = spotxBid.w;
+            bid.height = spotxBid.h;
+          }
+
+          if (request.video.ext.ad_unit == 'outstream') {
+            const renderer = Renderer.install({
+              id: 0,
+              url: '//',
+              config: {
+                adText: 'SpotX Outstream Video Ad via Prebid.js',
+                player_width: request.video.ext.player_width,
+                player_height: request.video.ext.player_height,
+                content_page_url: request.video.ext.content_page_url,
+                ad_mute: request.video.ext.ad_mute,
+                outstreamFunction: request.video.ext.outstreamFunction
+              }
+            });
+
+            try {
+              renderer.setRender(outstreamRender);
+              renderer.setEventHandlers({
+                impression: function impression() {
+                  return utils.logMessage('SpotX outstream video impression event');
+                },
+                loaded: function loaded() {
+                  return utils.logMessage('SpotX outstream video loaded event');
+                },
+                ended: function ended() {
+                  utils.logMessage('SpotX outstream renderer video event');
+                }
+              });
+            } catch (err) {
+              utils.logWarn('Prebid Error calling setRender or setEve,tHandlers on renderer', err);
+            }
+            bid.renderer = renderer;
+          }
+
+          bidResponses.push(bid);
+        })
+      });
+    }
+
+    // Make sure hb_adid will be correct
+    if (bidderRequest && bidderRequest.bidRequest && bidderRequest.bidRequest.bids) {
+      utils._each(bidderRequest.bidRequest.bids, function(spotxBid) {
+        let adId = null;
+        for (let i = 0; i < bidResponses.length; i++) {
+          if (bidResponses[i].requestId == spotxBid.bidId) {
+            adId = bidResponses[i].cache_key;
+          }
+        };
+        if (adId != null) {
+          spotxBid.bidId = adId;
+        }
+      });
+    }
+
+    return bidResponses;
+  },
+
+  /**
+   * If the publisher allows user-sync activity, the platform will call this function and the adapter may register pixels and/or iframe user syncs.
+   *
+   */
+  getUserSyncs: function(syncOptions, serverResponses) {},
+
+  /**
+   * If the adapter timed out for an auction, the platform will call this function and the adapter may register timeout.
+   *
+   */
+  onTimeout: function(timeoutData) {}
```

Comment:
This method is optional. Can be removed.

Reply:
removed